### PR TITLE
mesheryctl: Move constants to config object/file

### LINKED ISSUE #1022: [mesheryctl] Move constants to client configuration file
**Current Behavior**
`mesheryAuthToken` and `mesheryURL` are defined as constants in the `mesheryctl perf` command. Default values of both include reference to "http://localhost", assuming that the Meshery server will be locally deployed. It will often be the case the Meshery server is not deploye locally, however.

**Desired Behavior**
`mesheryAuthToken` and `mesheryURL` should be user configurable.

**Implementation / Acceptance Tests**
- [ ]  Values for these two variables should carry the same defaults that they do today (using "localhost").
- [ ]  `.meshery/mesheryctl.yaml` should serve as the Meshery Client configuration file.
- [ ]  `--ctlconfig` should be available as a persistent flag (under `perf` command and later globally under other commands) to allow users to override the these variables with alternative settings.

---
**Additional Context**
See discussion - https://github.com/layer5io/meshery/pull/1021/files/27dac1d18b235b59b39293ba111109a830ab4aff#diff-13173ea6e90ec366fe2b4d11e1cba8faR37

### PR DESCRIPTION
**Pull request by nitishm**
- Mesheryctl default configuration (endpoints, server URLs, etc) can now
be overriden by the user through a mesheryctl.yaml config file.
- Added debug flag for better reporting of mesheryctl commands

Fixes #1022
Signed-off-by: nitish.malhotra@gmail.com <nitish.malhotra@gmail.com>

**Description**

This PR fixes # 

**Notes to reviewers**

#### Checklist

- [] [Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin).
- [] PR title includes `[<name of area>]:<description>`

### CODE REVIEW

File: mesheryctl/internal/cli/root/root.go [hunk 1]
```diff
@@ -136,13 +148,46 @@ func initConfig() {
...
 	}
 
 	viper.AutomaticEnv() // read in environment variables that match
 
 	// If a config file is found, read it in.
 	if err := viper.ReadInConfig(); err == nil {
-		fmt.Println("Using config file:", viper.ConfigFileUsed())
+		log.Debugf("Using config file: %s", viper.ConfigFileUsed())
+	} else {
+		log.WithError(err).Errorf("failed to read in meshery configv")
```
Comment (kushthedude) [inline mesheryctl/internal/cli/root/root.go:160]: Is the v in the end typo ?

Comment (nitishm) [reply mesheryctl/internal/cli/root/root.go:160]: ah good catch. Will fix it.

**Comment by thakursaurabh1998** [comment]
That's a good thing done @nitishm 💥

**Comment by leecalcote** [comment]
What the? 😲 Why don’t my PRs get comments like this @kushthedude, @thakursaurabh1998?! 😠 

What’s so special about @nitishm’s code anyway?! What’s his code have that mine doesn’t? Mutiny!

**Comment by Nikhil-Ladha** [comment]
> What the?  Why don’t my PRs get comments like this @kushthedude, @thakursaurabh1998?! 
> 
> What’s so special about @nitishm’s code anyway?! What’s his code have that mine doesn’t? Mutiny!

Seems like, someone is in a very good mood and excited!🎉😆.

**Comment by thakursaurabh1998** [comment]
@leecalcote Maybe because this isn't a feature PR, but a PR which maintains the code sanity. And writing a code for making it cleaner hits different.

**Comment by kushthedude** [comment]
> What the? 😲 Why don’t my PRs get comments like this @kushthedude, @thakursaurabh1998?! 😠
> 
> What’s so special about @nitishm’s code anyway?! What’s his code have that mine doesn’t? Mutiny!

I am going to opt for special code classes from @nitishm so that my PRs can get such comments too 😄

**Comment by nitishm** [comment]
> > What the? 😲 Why don’t my PRs get comments like this @kushthedude, @thakursaurabh1998?! 😠
> > What’s so special about @nitishm’s code anyway?! What’s his code have that mine doesn’t? Mutiny!
> 
> I am going to opt for special code classes from @nitishm so that my PRs can get such comments too 😄

Stop it! you guys are making me blush 😊😊😊😊! With that said, @kushthedude I charge $50 an hour for in-person training 😉 😜 

> @leecalcote Maybe because this isn't a feature PR, but a PR which maintains the code sanity. And writing a code for making it cleaner hits different.

Agreed! Getting rid of code smell trumps new features any day 😉

**Comment by leecalcote** [comment]
This is really great. This will make for a nice, new section in the [Using mesheryctl](https://meshery.layer5.io/docs/guides/mesheryctl) guide

**Pull request merged**